code/game/objects/structures/maintenance.dm

```diff
@@ -0,0 +1,101 @@
+					/obj/item/restraints/handcuffs/cable/pink = 1,
+					/obj/item/restraints/handcuffs/alien = 2,
+					/obj/item/coin/bananium = 9,
+					/obj/item/kitchen/knife/butcher = 5,
+					/obj/item/coin/mythril = 1) //the loot table isn't that great and should probably be improved and expanded later.
+
+
+/obj/structure/moisture_trap/Initialize()
+	. = ..()
+	if(prob(40))
+		critter_infested = FALSE
+	if(prob(75))
+		var/picked_item = pickweight(loot)
+		hidden_item = new picked_item(src)
+	loot = null
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_MOIST, CELL_VIRUS_TABLE_GENERIC, rand(2,4), 20)
+
+/obj/structure/moisture_trap/Destroy()
+	if(hidden_item)
+		QDEL_NULL(hidden_item)
+	return ..()
+
+///This proc checks if we are able to reach inside the trap to interact with it.
+/obj/structure/moisture_trap/proc/CanReachInside(mob/user)
+	if(!isliving(user))
+		return FALSE
+	var/mob/living/L = user
+	if((L.mobility_flags & MOBILITY_STAND) && ishuman(L)) //I dont think monkeys can crawl on command.
+		return FALSE
+	return TRUE
+
+/obj/structure/moisture_trap/attack_hand(mob/user)
+	. = ..()
+	if(iscyborg(user) || isalien(user))
+		return
+	if(!CanReachInside(user))
+		to_chat(user, "<span class='warning'>You need to lie down to reach into [src].</span>")
+		return
+	to_chat(user, "<span class='notice'>You reach down into the cold water of the basin.</span>")
+	if(!do_after(user, 20, target = src))
+		return
+	if(hidden_item)
+		user.put_in_hands(hidden_item)
+		to_chat(user, "<span class='notice'>As you poke around inside [src] you feel the contours of something hidden below the murky waters.</span>\n<span class='nicegreen'>You retrieve [hidden_item] from [src].</span>")
+		hidden_item = null
+		return
+	if(critter_infested && prob(50) && iscarbon(user))
+		var/mob/living/carbon/bite_victim = user
+		var/obj/item/bodypart/affecting = bite_victim.get_bodypart("[(user.active_hand_index % 2 == 0) ? "r" : "l" ]_arm")
+		if(affecting?.receive_damage(30))
+
+			to_chat(user, "<span class='danger'>You feel a sharp as an unseen creature sinks it's [pick("fangs", "beak", "proboscis")] into your arm!</span>")
+			bite_victim.update_damage_overlays()
+			playsound(src,'sound/weapons/bite.ogg', 70, TRUE)
+			return
+	to_chat(user, "<span class='warning'>You find nothing of value...</span>")
+
+/obj/structure/moisture_trap/attackby(obj/item/I, mob/user, params)
+	if(iscyborg(user) || isalien(user) || !CanReachInside(user))
+		return ..()
+	add_fingerprint(user)
+	if(istype(I, /obj/item/reagent_containers))
+		if(istype(I, /obj/item/reagent_containers/food/snacks/monkeycube))
+			var/obj/item/reagent_containers/food/snacks/monkeycube/cube = I
+			cube.Expand()
+			return
+		var/obj/item/reagent_containers/RG = I
+		if(RG.is_open_container())
+			RG.reagents.add_reagent(/datum/reagent/water, min(RG.volume - RG.reagents.total_volume, RG.amount_per_transfer_from_this))
+			to_chat(user, "<span class='notice'>You fill [RG] from [src].</span>")
```

Comment:
* Better var names, getting tired of writing suggestions, just stop coding in this pattern

Reply:
this code is months old, there was no such contribution guideline back then, this is something you started requesting after this PR was made so the attitude isn't really necessary.

Reply:
Fair enough